### PR TITLE
Add testing distribution and sections

### PR DESCRIPTION
**Pull request by rybaktomasz**
This change adds ability to create images for testing, and to add contrib and non-free sections. The latter will be especially useful for GPGPU instances.

### CODE REVIEW

File: base/manifest-schema.json [hunk 1]
```diff
@@ -25,7 +25,8 @@
 		"system": {
 			"type": "object",
 			"properties": {
-				"release":      { "enum": ["wheezy", "unstable"] },
+				"release":      { "enum": ["wheezy", "testing", "unstable"] },
+				"sections":     { "type": "array" },
```
Comment (andsens) [inline base/manifest-schema.json]: You should probably add a `"minItems": 1` here, otherwise you can create an invalid apt source.

**Comment by andsens** [comment]
Great PR! I'll merge once you added the `minItems` thing.

**Comment by rybaktomasz** [comment]
Added; Code in apt.py ensures that section "main" is present if no section is given in manifest, but this'll make sure that array with sections is not empty.

**Comment by andsens** [comment]
perfect! :-)

**Pull request merged**